_posts/2021-03-10-katib-0.10.md

```diff
@@ -0,0 +1,310 @@
+their ML infrastructure pipeline.
+
+## System Architecture
+
+<img src="/images/2021-03-10-katib-0.10/architecture.png" width="" alt="Architecture" title="">
+
+There are 3 main concepts in Katib which are Kubernetes CRDs:
+
+- **Experiment** - a single optimization run with objective, search space, and
+  search algorithm.
+
+- **Suggestion** - set of hyperparameters, which are produced by a user's
+  selected search algorithm. Katib creates Trials to evaluate them.
+
+- **Trial** - one iteration of the hyperparameters tuning process. Trial runs
+  the worker job which corresponds to the training job. Since Trial is an
+  abstraction on top of the worker job, any Kubernetes resource can be used to
+  perform the training job. For example, [**TFJob**](https://github.com/kubeflow/tf-operator),
+  [**MPIJob**](https://github.com/kubeflow/mpi-operator) or even
+  [Tekton **Pipeline**](https://github.com/tektoncd/pipeline).
+
+By using above resources Katib follows the following steps, which are marked
+in the diagram above:
+
+1. Once an Experiment is submitted, the Experiment controller creates an
+   appropriate Suggestion object.
+
+1. The Suggestion controller creates an AutoML algorithm service based on this
+   Suggestion object. When the algorithm service is ready, the Suggestion controller
+   calls the service to get new parameters and appends them to the Suggestion object.
+
+1. The Experiment controller finds that Suggestion object has been updated and
+   creates a corresponding Trial object for each set of parameters.
+
+1. The Trial controller generates a worker job for each Trial object and watches
+   for the status of each job. The worker job based on the Trial template.
+
+1. Once the worker job has been completed, the metrics collector gets the metrics
+   from the job and persists them in the database.
+
+1. The Experiment controller sends the metrics results to the algorithm service
+   and gets new parameters from the Suggestion object.
+
+## Custom Kubernetes resources support
+
+The latest Katib version 0.10 implements a new feature to support any Kubernetes
+CRDs or Kubernetes workloads as a Katib Trial template. Therefore, there is no
+need to manually modify the Katib controller to use CRD as a Trial. As long as
+the CRD creates Kubernetes Pods, allows injecting the
+[sidecar container](https://kubernetes.io/docs/concepts/workloads/pods/)
+on these Pods, and has success and failure status, the CRD can be used in Katib.
+
+The motivation behind this feature are as follows:
+
+- Katib Trial template supports only a limited type of Kubernetes resources
+  ([**BatchJob**](https://kubernetes.io/docs/concepts/workloads/controllers/job/),
+  [**TFJob**](https://www.kubeflow.org/docs/components/training/tftraining/) and
+  [**PyTorchJob**](https://www.kubeflow.org/docs/components/training/pytorch/)).
+
+- Many Katib users might have their own CRDs which they want to use as a Trial template.
```

Comment:
Can we standardize to either "Trial Template" or "Trial template" consistently through the post?